.github/workflows/e2e-suite.yml

```diff
@@ -1,12 +1,10 @@
-name: E2E Test Suite
+name: Integration Tests
 on:
   workflow_dispatch: null
   push:
-    branches:
-      - main
-      - dev
+  pull_request:
```

Comment:
For security purposes, this workflow should probably only run on `push` rather than on `push` and `pull_request`. 

To address this, we have a [workflow](https://github.com/linode/linode-cli/blob/dev/.github/workflows/e2e-suite-pr-command.yml) to manually trigger E2E tests when a maintainer uses the `/acctest` command. This is necessary as an external contributor could potentially push up malicious code to grab our test account token, etc. without maintainer approval. Fortunately, GitHub Actions only exposes secrets to workflows that explicitly use them, so our other workflows shouldn't be susceptible to this.